src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -1138,8 +1138,11 @@ void Leader::RemoveRlocInPrefix(PrefixTlv       &aPrefix,
     {
         if (aPrefix.GetSubTlvsLength() == sizeof(ContextTlv))
         {
-            context->ClearCompress();
-            StartContextReuseTimer(context->GetContextId());
+	    if (context->IsCompress())
```

Comment:
Thanks for catching this.

I think a better approach is to avoid this entire block of code (starting from L1137) if we never called `RemoveTlv()` above. We could set a bool when calling `RemoveTlv()` and use `VerifyOrExit()` before L1137 above.

Reply:
Hi @jwhui Thanks for the feedback. 

If RemoveTlv is invoked it would mean either of the two cases right 

 1. Prefix had HasRoute subTLV which was removed (in which case it wouldn't have a 6LoWPAN ID TLV)
 2. Prefix had BorderRouter subTLV which was removed (in which cast it would still have a 6LoWPAN ID TLV)

In case-1 above we could avoid the block of code (starting from L1137) but In case-2 we cannot avoid right (It needs to clear the ClearCompress flag and StartContextReuseTimer for the first time?

Wondering if just checking IsCompress (as in this fix)  would be cleaner? Do you think it might break any scenario? 
Thanks!

Reply:
In general, a prefix is either on-mesh or external route, never both. Context IDs are only added for on-mesh prefixes. So the only time we should be starting the timer is after removing a Border Router TLV.

After chatting with @abtink , I think it makes sense for the quick fix to simply check if the timer is already set and, if so, avoid setting it again. @abtink will submit a PR with the proposed change soon.